src/_pytest/deprecated.py

```diff
@@ -101,6 +101,11 @@
     "see https://docs.pytest.org/en/latest/deprecations.html"
     "#py-path-local-arguments-for-hooks-replaced-with-pathlib-path",
 )
+
+WARNS_NONE_ARG = PytestDeprecationWarning(
+    "Please pass an explicit Warning type or tuple of Warning types."
```

Comment:
I see this warning is triggered when users use `pytest.warn(None)`, which was valid and meant to catch any warning, so I think this messages would better help users to explicitly mention that deprecation:

```suggestion
    "Passing None to catch any warning has been deprecated, pass no arguments instead:\n"
    "  Replace pytest.warns(None) by simply pytest.warns()."
```